ui/hooks/bridge/useTokensWithFiltering.ts

```diff
@@ -241,10 +306,15 @@ export const useTokensWithFiltering = (
               token.chainId,
             )
           ) {
-            if (
-              (isNativeAddress(token.address) || token.isNative) &&
-              !isSolanaChainId(token.chainId)
-            ) {
+            if (isNativeAddress(token.address) || token.isNative) {
+              // Skip native tokens for EVM chains as they were already yielded above
+              if (
+                !isSolanaChainId(chainId) &&
+                !isBitcoinChainId(chainId) &&
+                token.chainId === chainId
+              ) {
+                continue;
+              }
```

Comment:
I believe you won't need this either if you remove the block above

```suggestion
              if (
                !isSolanaChainId(chainId) &&
                !isBitcoinChainId(chainId) &&
                token.chainId === chainId
              ) {
                continue;
              }
```